server/apiv2/handlers/tso_keyspace_group.go

```diff
@@ -164,8 +170,21 @@ func SplitKeyspaceGroupByID(c *gin.Context) {
 	}
 
 	svr := c.MustGet(middlewares.ServerContextKey).(*server.Server)
-	manager := svr.GetKeyspaceGroupManager()
-	err = manager.SplitKeyspaceGroupByID(id, splitParams.NewID, splitParams.Keyspaces)
+	patrolKeyspaceAssignmentState.Lock()
+	if !patrolKeyspaceAssignmentState.patrolled {
+		// Patrol keyspace assignment before splitting keyspace group.
+		manager := svr.GetKeyspaceManager()
+		err = manager.PatrolKeyspaceAssignment()
+		if err != nil {
```

Comment:
if meet err returned, the lock will hang?

Reply:
Nice catch. Fixed.